platforms/core-runtime/java-language-extensions/src/main/java/org/gradle/api/JavaVersion.java

```diff
@@ -310,25 +310,8 @@ public String toString() {
         return versionName;
     }
 
-    /**
-     * Returns the major version number as a {@link String}.
-     *
-     * @see #getMajorVersionNumber() for the integer value
-     * @since 1.8
-     */
     public String getMajorVersion() {
-        return String.valueOf(getMajorVersionNumber());
-    }
-
-    /**
-     * Returns the major version number.
-     *
-     * @see #getMajorVersion() for the string value
-     * @since 8.9
-     */
-    @Incubating
-    public int getMajorVersionNumber() {
```

Comment:
🤔 I wonder if it would make sense to leave in the new method, so at some point we could use it. Though maybe that is too optimistic and too dangerous.
Could you add a test that exposes the problem we experienced, so it doesn't happen again?

Reply:
Based on the fact that even without announcement people picked up this method signifies to me that there is value to this method, and I would try to see what we could do.

> Could you add a test that exposes the problem we experienced, so it doesn't happen again?

https://github.com/gradle/gradle/issues/29606